lib/apiservers/engine/backends/system.go

```diff
@@ -288,6 +288,10 @@ func (s *System) AuthenticateToRegistry(ctx context.Context, authConfig *types.A
 		registryAddress = registryAddress + "/v2/"
 	}
 
+	if !strings.HasPrefix(registryAddress, "https://") && !strings.HasPrefix(registryAddress, "http://") {
+		registryAddress = "https://" + registryAddress
```

Comment:
Yeah without writing our own custom parser this is always going to bite us though. I got around this in my nfs-volume-vic-machine work by doing the url.Parse and then checking all the fields I cared about for sane inputs then rebuilding the string again. it looked awful, but the other option is to break out the regex.